src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts

```diff
@@ -227,15 +237,67 @@ class WelcomePage extends Disposable {
 			name: localize('welcome.title', "Welcome"),
 			resource,
 			telemetryFrom,
-			onReady: (container: HTMLElement) => this.onReady(container, recentlyOpened, installedExtensions, fileService)
+			onReady: (container: HTMLElement) => this.onReady(container, recentlyOpened, installedExtensions, fileService, layoutService)
 		});
 	}
 	public openEditor() {
 		return this.editorService.openEditor(this.editorInput, { pinned: false });
 	}
-	private onReady(container: HTMLElement, recentlyOpened: Promise<IRecentlyOpened>, installedExtensions: Promise<IExtensionStatus[]>, fileService: IFileService): void {
+	private onReady(container: HTMLElement, recentlyOpened: Promise<IRecentlyOpened>, installedExtensions: Promise<IExtensionStatus[]>, fileService: IFileService, layoutService: ILayoutService, ): void {
 		const enabled = isWelcomePageEnabled(this.configurationService, this.contextService);
 		const showOnStartup = <HTMLInputElement>container.querySelector('#showOnStartup');
+		const guidedTourEnabled = isGuidedTourEnabled(this.configurationService);
+		if (enabled) {
+			showOnStartup.setAttribute('checked', 'checked');
+		}
+
+		if (guidedTourEnabled) {
```

Comment:
Can this logic here live in its own method? This onReady method seems to be getting large

Reply:
I believe this has been fixed